lib/endpoint-resolver.js

```diff
@@ -0,0 +1,171 @@
+
+const routerFetcher = new RouterFetcher();
+
+class EndpointResolver {
+	constructor(service) {
+		this.service = service;
+		this.endpointResolverLocal = new EndpointResolverLocal('schemas/public.json');
+		this.requestCache = {};
+	}
+
+
+	/**
+	 * Helper for check if a object is a endpoint source schema valid
+	 * @param {Object} obj
+	 * @returns {Boolean}
+	 */
+	static isEndpointSource(obj) {
+		return ajv.validate(obj, {
+			$ref: 'schemaDefinitions#/definitions/endpoint'
+		});
+	}
+
+
+	/**
+	 * Helper for find resolve endpoint data
+	 * @param {Object} currentEndpoint
+	 * @param {Object} endpointResolved
+	 */
+	static getEndpointResolved(currentEndpoint, endpointResolved) {
+		const { service, namespace, method } = endpointResolved;
+		return isMatch(currentEndpoint, { service, namespace, method }) ? endpointResolved : false;
+	}
+
+
+	/**
+	 * Helper for call router fetcher for get endpoint data
+	 * @param {string} service
+	 * @param {string} namespace
+	 * @param {string} method
+	 * @return {<Promise>}
+	 */
+	callFetcher(service, namespace, method) {
+		let promise;
+
+		if(this.service && this.service === service)
+			promise = this.endpointResolverLocal.resolve(namespace, method);
+		else
+			promise = routerFetcher.getEndpoint(service, namespace, method);
+
+		// Save promise in requestCache
+		objSet(this.requestCache, `${service}.${namespace}.${method}`, promise);
+
+		return promise;
+	}
+
+
+	/**
+	 * Helper for resolve promises for get endpoints
+	 * @param {Object}
+	 * @returns {<Promise>}
+	 */
+	async getEndpoint({ service, namespace, method }) {
+		// Find promise in requestCache for not call again this.callFetcher
+		const promise = objGet(this.requestCache, `${service}.${namespace}.${method}`);
+
+		try {
+			const response = await (promise || this.callFetcher(service, namespace, method));
+			const { url, endpoint, httpMethod } = response;
+
+			return {
+				service,
+				namespace,
+				method,
+				url: endpoint || url,
+				httpMethod
+			};
+
+		} catch(err) {
+			const error = new Error(`Fail call routerFetcher for ${JSON.stringify({ service, namespace, method })}`);
+			error.errors = [err];
+			throw error;
+		}
+
+	}
+
+
+	/**
+	 * Add resolve data in all endpoints into schema
+	 * @param {Array} endpointsToResolve
+	 * @param {Array} endpointsResolved
+	 */
+	addResolveDataToEndpoints(endpointsToResolve, endpointsResolved) {
+		endpointsToResolve.forEach(currentEndpoint => {
+			const endpoint = endpointsResolved.find(endpointResolved => (
+				EndpointResolver.getEndpointResolved(currentEndpoint, endpointResolved)
+			));
+
+			const { httpMethod, url } = endpoint;
+			currentEndpoint.httpMethod = httpMethod.toLowerCase();
+			currentEndpoint.url = url;
+		});
+	}
+
+
+	/**
+	 * Resolve all endpoints found in the schema.
+	 * @param  {Object} schema The schema.
+	 * @return {<Promise>} Array with all endpoints resolved.
+	*/
+	resolveEndpoints(endpointsToResolve) {
```

Comment:
Ya podrías hacer todo de un _raquetazo_..
Algo asi: 
```suggestion
	resolveEndpoints(endpointsToResolve) {

		// Con esto ya getteas el endpoint (que se hace una sola vez por la cache)
		// Cuando esa promise resuelva, va a reemplazar la info (sin recorrer nada, ya tiene todo lo que necesita)
		// Finalmente devuelve una nueva promise que awaiteás ahi abajo con el Promise.all()
		// PD: falta `addResolvedData()`, pero es muy parecido a `addResolveDataToEndpoints()` sin el foreEach ni el find.
		const calls = endpointsToResolve.map(endpoint => {
				return this.getEndpoint(endpoint)
					.then(resolvedEndpoint => this.addResolvedData(endpoint, resolvedEndpoint));
			});

		return Promise.all(calls);
	}
```